blog/_posts/2024-08-23-fine-tuning-language-models-with-kaito.md

```diff
@@ -0,0 +1,145 @@
+---
+title: "Fine tune language models with KAITO on AKS"
+description: "Make your AI model even smarter, in just a few steps"
+date: 2024-08-23
+author: Sachi Desai
+categories:
+- ai
+- add-ons
+tags:
+- kaito
+---
+
+## Introduction
+
+You may have heard of the [Kubernetes AI Toolchain Operator](https://github.com/Azure/kaito/tree/main) (KAITO) announced at Ignite 2023 and KubeCon Europe this year. The open source project has gained popularity in recent months by introducing a streamlined approach to AI model deployment and flexible infrastructure provisioning on Kubernetes.
+
+With the [v0.3.0 release](https://github.com/Azure/kaito/releases/tag/v0.3.0), KAITO has expanded the supported model library to include the Phi-3 model, but the biggest (and most exciting) addition is the ability to fine-tune open-source models. Why should you be excited about fine-tuning? Well, it’s because fine-tuning is one way of giving your foundation model additional training using a specific dataset to enhance accuracy, which ultimately improves the interaction with end-users. (Another way to increase model accuracy is Retrieval-Augmented Generation (RAG), which we touch on briefly in this [section](#Stay-tuned-to-make-models-smarter), coming soon to KAITO).
+
+## Fine-tuning a foundation model is sometimes necessary
+
+Let’s experiment with [Phi-3-Mini-128K-Instruct](https://huggingface.co/microsoft/Phi-3-mini-128k-instruct), a robust and high-performing model that’s a bit smaller in size than your average LLM. We’ll use a custom [Streamlit](https://streamlit.io/) chat UI app, connected to a Phi-3 inference service deployed by KAITO, and ask a basic question about AKS:
+
+![Screenshot of untuned model response](/AKS/assets/images/fine-tuning-language-models-with-kaito/untuned-lm-with-kaito.png)
+_Screenshot of un-tuned model response_
+
+Uh oh! That is **NOT** the answer we were expecting – we’d expect the model to know more about Kubernetes and cloud technologies. This foundation model was trained on a mix of synthetic data and public websites, so it doesn’t recognize AKS as Azure Kubernetes Service, and got creative in answering the question to the best of its ability. This is a well-known problem in the AI community and various fine-tuning techniques have been invented to enhance model accuracy using domain-specific knowledge. In our specific use case, we need the Phi-3 model to have more knowledge of Kubernetes and cloud platforms.
+
+## KAITO can help with model fine-tuning
+
+KAITO supports Low-Rank Adaption (LoRA) and Quantized LoRA (QLoRA), the [most common](https://arxiv.org/pdf/2402.16141) parameter-efficient fine-tuning techniques, which significantly lower the GPU requirement for tuning a language model. With LoRA, fine-tuning an LLM in a single GPU is possible! 
```

Comment:
"significantly lowers"

Reply:
Both techniques lower the GPU requirement for tuning, so I kept it plural